akka-http/src/main/scala/akka/http/scaladsl/marshalling/PredefinedToResponseMarshallers.scala

```diff
@@ -105,24 +106,13 @@ trait LowPriorityToResponseMarshallerImplicits {
           val availableMarshallingsPerElement = source.mapAsync(1) { t ⇒ m(t)(ec) }
 
           // TODO optimise such that we pick the optimal marshalling only once (headAndTail needed?)
```

Comment:
I can't really see how this could be achievable. So long as `m.apply` returns a `List[Marshalling[_]]` for each input, it seems inevitable that it'll have to go through the list and find the first adequate marshaller. For that purpose, the code below looks as efficient as could be.

Reply:
Agreed. The issue is that the current generic streaming support doesn't seem to be as useful as we thought initially. The same functionality would probably be simpler and faster if implemented directly in the target marshallers. For example, we could provide a `ToEntityMarshaller[Source[T]]` in akka-http-marshallers-spray-json directly.

Reply:
Okay; I've removed the TODO - is that right?

Reply:
That's fine